File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        mySet = set(nums)

        # check the beginning of the sequence
        maxCount = 0
        for number in mySet:
            if (number-1) not in mySet:
                count = 1
                targetNumber = number + 1
                
                while targetNumber in mySet:
                    targetNumber += 1
                    count += 1

                maxCount = max(maxCount, count)
                
        return maxCount
            

File: Data Structures & Algorithms/max-consecutive-ones/submission-1.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int findMaxConsecutiveOnes(std::vector<int>& nums) {
        int current_occurrence_1 = 0;
        int max_occurrence_1 = 0;

        for (int number : nums)
        {
            if (number == 0) {current_occurrence_1 = 0;}
            else {current_occurrence_1 += 1;}

            if (max_occurrence_1 < current_occurrence_1) 
            {max_occurrence_1 = current_occurrence_1;}
        }

        return max_occurrence_1;
    }
};